stellar_sdk/soroban_server_async.py

```diff
@@ -235,6 +236,37 @@ async def send_transaction(
         )
         return await self._post(request, SendTransactionResponse)
 
+    async def poll_transaction(self, transaction_hash: str, attempts: int = 30, sleep_strategy: int = 1) -> GetTransactionResponse:
+        """Poll for a particular transaction with certain parameters.
+
+        After submitting a transaction, clients can use this to poll for transaction completion and return a definitive state of success or failure.
+
+        :param transaction_hash: The hash of the transaction to poll for.
+        :type transaction_hash: str
+        :param attempts: The number of attempts to make before returning the last-seen status, defaults to 30.
+        :type attempts: int, optional
+        :param sleep_strategy: The amount of time, in seconds, to wait for between each attempt, defaults to 1.
+        :type sleep_strategy: int, optional
+        :return: A :class:`GetTransactionResponse <stellar_sdk.soroban_rpc.GetTransactionResponse>` response object after a "found" response, (which may be success or failure) or the last response obtained after polling the maximum number of specified attempts.
+        :rtype: GetTransactionResponse
+        """
+        # positive and defined user value or default
+        max_attempts: int = 30 if (attempts or 0) < 1 else attempts or 30
+
+        attempt: int = 0
+        resp = await self.get_transaction(transaction_hash=transaction_hash)
+
+        while attempt < max_attempts:
+            if (resp.status != GetTransactionStatus.NOT_FOUND):
+                return resp
+
+            attempt += 1
+            resp = await self.get_transaction(transaction_hash=transaction_hash)
+
+            time.sleep(sleep_strategy or 1)
```

Comment:
Here we need the async version of sleep to not block the event loop. (Needs the `import asyncio`)
```suggestion
            asyncio.sleep(sleep_strategy or 1)
```

Reply:
Or, maybe we would need to `await asyncio.sleep(...)`?

Reply:
Ark yes sorry my suggestion was indeed missing the `await`.

To complement what @overcat said, in Python, asynchronous code is running on a single thread and a there is only a single thread running at a given time (things are changing there, but another story.) So if you have a backend processing multiple requests to e.g. submit some transactions. When you process a request, you might want to tell the event handler that while you are waiting `sleep_strategy`, the backend can handle another request.

Reply:
Thank you both for helping me understand this better! I feel like my eyes are just beginning to open to the wide/wonderful world of async python, and it's probably much cooler than I thought it would be!